CMakeLists.txt

```
@@ -1,5 +1,9 @@
 cmake_minimum_required(VERSION 3.13)
 
+if (CMAKE_CURRENT_BINARY_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  message(FATAL_ERROR "Building in root or relevant directories will cause issues.\nPlease use a dedicated build directory and remove the CMakeCache.txt file and CMakeFiles dir created in the root.")
```

```suggestion
  message(FATAL_ERROR "Celerity does not support in-source builds.\nPlease use a dedicated build directory and remove ${CMAKE_SOURCE_DIR}/CMakeCache.txt and ${CMAKE_SOURCE_DIR}/CMakeFiles.")
```

I think we should mention the term "in-source build" explicitly here. Also we can print the exact path, as "root" is ambiguous if Celerity is built as a subproject.